src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentDocumentClassifierPass.cs

```diff
@@ -1,18 +1,27 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.AspNetCore.Razor.Language;
-using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
-using Microsoft.AspNetCore.Razor.Language.Intermediate;
 using System;
 using System.IO;
 using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
 
 namespace Microsoft.AspNetCore.Blazor.Razor
 {
-    internal class ComponentDocumentClassifierPass : DocumentClassifierPassBase, IRazorDocumentClassifierPass
+    public class ComponentDocumentClassifierPass : DocumentClassifierPassBase, IRazorDocumentClassifierPass
     {
-        protected override string DocumentKind => "Blazor.Component-0.1";
+        public static readonly string ComponentDocumentKind = "Blazor.Component-0.0.5";
+
+        private static readonly char[] PathSeparators = new char[] { '/', '\\' };
+
+        // This will be set by our CLI tool when running at the command line. The default value
+        // will be used by the IDE.
+        public string BaseNamespace { get; set; } = "HostedInAspNet.Client"; //"__BlazorGenerated";
```

Comment:
This is the one sticky thing I ran into. Razor doesn't provide the base namespace today and it probably should.

I think it makes more sense to try and work around this rather than blocking on it. Let's discuss at the sync.